### PR TITLE
KEYCLOAK-7470 Ability to order client scopes

### PR DESCRIPTION
**Pull request by mposolda**

### CODE REVIEW

**Comment by mposolda** [comment]
@pedroigor @hmlnarik @stianst Anyone has a chance to review?

File: themes/src/main/resources/theme/base/admin/messages/admin-messages_en.properties [hunk 1]
```diff
@@ -827,6 +827,8 @@ client-scope.display-on-consent-screen=Display On Consent Screen
 client-scope.display-on-consent-screen.tooltip=If on, and this client scope is added to some client with consent required, then the text specified by 'Consent Screen Text' will be displayed on consent screen. If off, then this client scope won't be displayed on consent screen
 client-scope.consent-screen-text=Consent Screen Text
 client-scope.consent-screen-text.tooltip=Text, which will be shown on consent screen when this client scope is added to some client with consent required. Defaults to name of client scope if it's not filled
+client-scope.gui-order=GUI order
+client-scope.gui-order.tooltip=Number defining order of the provider in GUI (eg. on Consent page).
```
Comment (hmlnarik) [inline themes/src/main/resources/theme/base/admin/messages/admin-messages_en.properties]: Specify order of the provider in GUI (e.g. in Consent page) as integer

File: services/src/main/java/org/keycloak/forms/account/freemarker/model/ApplicationsBean.java [hunk 2]
```diff
@@ -72,18 +76,17 @@ public ApplicationsBean(KeycloakSession session, RealmModel realm, UserModel use
...
-
-                    for (ClientScopeModel clientScope : consent.getGrantedClientScopes()) {
-                        String consentText = clientScope.getConsentScreenText();
-                        clientScopesGranted.add(consentText);
-                    }
+                    orderedScopes.addAll(consent.getGrantedClientScopes());
                 }
             }
+            List<String> clientScopesGranted = orderedScopes.stream()
+                    .map(clientScope -> clientScope.getConsentScreenText())
```
[the remark this answers is not in the document]
Comment (mposolda) [reply services/src/main/java/org/keycloak/forms/account/freemarker/model/ApplicationsBean.java]: Good to know. Thanks for pointing this.

**Comment by mposolda** [comment]
@hmlnarik Addressed your feedback with the exception of the comparator. It can't return 0 unless objects are the same (equals). Because GUI Order is not mandatory field for the client scopes and identity providers, so  returning 0 would result that some object without GUIOrder specified, won't be put to the collection. Added one more unit test for this (IdentityProviderBeanTest.testIdentityProviderComparatorForEqualObjects)

**Comment by hmlnarik** [comment]
Reason is that `TreeSet` is used which does not allow duplicates. List should be used instead

**Comment by mposolda** [comment]
@hmlnarik I've changed everywhere to the List as we discussed

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**